services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -3526,13 +3526,32 @@ private OfflinePushStatusInfo getOffLineJobStatus(
       String storeName = Version.parseStoreFromKafkaTopicName(kafkaTopic);
       int versionNum = Version.parseVersionFromKafkaTopicName(kafkaTopic);
       boolean deferredSwap = isDeferredSwap(clusterName, storeName, versionNum);
+      HelixVeniceClusterResources resources = getVeniceHelixAdmin().getHelixVeniceClusterResources(clusterName);
+
       if (!deferredSwap) {
-        truncateTopicsOptionally(
-            clusterName,
-            kafkaTopic,
-            incrementalPushVersion,
-            currentReturnStatus,
-            currentReturnStatusDetails);
+        try (AutoCloseableLock ignore = resources.getClusterLockManager().createStoreWriteLock(storeName)) {
+          ReadWriteStoreRepository repository = resources.getStoreMetadataRepository();
+          Store parentStore = repository.getStore(storeName);
+          boolean isTargetRegionPush = !StringUtils.isEmpty(targetedRegions);
+          // Truncate topic after push is in terminal state if
+          // 1. Its a hybrid store or regular push
+          // 3. If target push is enabled and its previously pushed by previous colo push (status == PUSHED)
+          if (!isTargetRegionPush || parentStore.getVersion(versionNum).get().getStatus().equals(PUSHED)
```

Comment:
`PUSHED` means it finishes in all 3 colos?

Reply:
Here PUSHED means it pushed to the target colo and eventually to the other colo. If the push fails at any step, it will be updated to KILLED.